consumer/consumer.go

```diff
@@ -365,18 +392,38 @@ func (consumer *StateSyncerConsumer) readAndDecodeNextEntry(reader *bufio.Reader
 		}
 		return nil, true, nil
 	} else if err != nil {
+		// Reset the reader to the position before the unexpected EOF.
+		if _, err = file.Seek(currentPos, io.SeekStart); err != nil {
+			return nil, false, errors.Wrapf(err, "consumer.readAndDecodeNextEntry: Error seeking to current position in file")
+		}
```

Comment:
maybe we write an internal helper function here to handle this seek to seek start position? seems like we're copying this code block a lot.

Reply:
I thought about this originally, but then I realized most of the code here would remain. I suppose we could still write a file.Seek wrapper and have the io.SeekStart hardcoded, but otherwise this code block will likely remain largely unchanged. Happy to implement that if you feel it's a better coding standard, personally I could go either way.

Reply:
Just leave it - it doesn't get much better with the refactor as you stated.